src/Core/Config/Cache/ConfigCacheManager.php

```diff
@@ -214,4 +220,179 @@ private function loadConfigFile($filepath)
+	 * @param string $value The new value
+	 */
+	public function saveToConfigFile($cat, $key, $value)
+	{
+		$this->saveToLegacyConfig('htpreconfig', $cat, $key, $value);
+		$this->saveToLegacyConfig('htconfig', $cat, $key, $value);
+
+		$this->saveToCoreConfig('local', $cat, $key, $value);
+	}
+
+	/**
+	 * Saves a value to either an config or an ini file
+	 *
+	 * @param string $name  The configuration file name ('local', 'addon', ..)
+	 * @param string $cat   The configuration category
+	 * @param string $key   The configuration key
+	 * @param string $value The new value
+	 */
+	private function saveToCoreConfig($name, $cat, $key, $value)
+	{
+		if (file_exists($this->configDir . DIRECTORY_SEPARATOR . $name . '.config.php')) {
+			$this->saveConfigFile($this->configDir . DIRECTORY_SEPARATOR . $name . '.config.php', $cat, $key, $value);
+		} elseif (file_exists($this->configDir . DIRECTORY_SEPARATOR . $name . '.ini.php')) {
+			$this->saveINIConfigFile($this->configDir . DIRECTORY_SEPARATOR . $name . '.ini.php', $cat, $key, $value);
+		} else {
+			return;
+		}
+	}
+
+	/**
+	 * Saves a value to a config file
+	 *
+	 * @param string $fullName The configuration full name (including the path)
+	 * @param string $cat   The configuration category
+	 * @param string $key   The configuration key
+	 * @param string $value The new value
+	 */
+	private function saveConfigFile($fullName, $cat, $key, $value)
+	{
+		$reading = fopen($fullName, 'r');
+		$writing = fopen($fullName . '.tmp', 'w');
+
+		$categoryFound = false;
+		$categoryBracketFound = false;
+		$lineFound = false;
+		$lineArrowFound = false;
+
+		while (!feof($reading)) {
+			$line = fgets($reading);
+
+			// find lines like '
+			if (!$categoryFound && stristr($line, sprintf('\'%s\'', $cat))) {
+				$categoryFound = true;
+			}
+
+			if ($categoryFound && !$categoryBracketFound && stristr($line, '[')) {
+				$categoryBracketFound = true;
+			}
+
+			if ($categoryBracketFound && !$lineFound && stristr($line, sprintf('\'%s\'', $key))) {
+				$lineFound = true;
+			}
+
+			if ($lineFound && !$lineArrowFound && stristr($line, '=>')) {
+				$lineArrowFound = true;
+			}
+
+			if ($lineArrowFound && preg_match_all('/\'(.*?)\'/', $line, $matches, PREG_SET_ORDER)) {
+				$lineVal = end($matches)[0];
+				$writeLine = str_replace($lineVal, '\'' . $value . '\'', $line);
+				$categoryFound = false;
+				$categoryBracketFound = false;
+				$lineFound = false;
+				$lineArrowFound = false;
+			} elseif ($categoryBracketFound && !$lineArrowFound && stristr($line, ']')) {
+				$categoryFound = false;
+				$categoryBracketFound = false;
+				$lineFound = false;
+				$lineArrowFound = false;
+				$writeLine = sprintf(self::IDENT . self::IDENT .'\'%s\' => \'%s\',' . PHP_EOL, $key, $value);
+				$writeLine .= $line;
+			} else {
+				$writeLine = $line;
+			}
+
+			fputs($writing, $writeLine);
+		}
+
+		fclose($reading);
+		fclose($writing);
+		rename($fullName . '.tmp', $fullName);
```

Comment:
mhm .. currently if I call this function twice, the rename will be called twice too.. So currently I would just preserve the latest change of all changes in the `.old` file.

Shall I alter this function to save all changes at once?

Reply:
BTW: Have you considered the situation when the system doesn't have write access?

Reply:
> Shall I alter this function to save all changes at once?

Yes, I think it would be better to have all the changes made at once to the config file so that a backup can be kept.

Reply:
@annando currently I'm thinking of throwing an exception, catching it during the update process and stop the update. But I'm unsure if we should skip the whole update so the Build release remains on the old version or just print a hint on the admin panel

@MrPetovan allright :-)

Reply:
Throwing an exception is the best way to confuse the users ;-) I would prefer pinning a huge information on the admin page to suggest the steps that need to be done.

Reply:
I don't want to throw an exception to the frontend :D . But it is possible that we don't have a working "Logger" instance inside the `ConfigCacheManager`, so I'd like to throw the exception and catch it inside the `update_1303()` and decide if we can Log it.

And of course printing a big information at least.

The question is, if I have to completely block the whole update if we cannot updatee or if I have "just" to add a warning to the admin panel.